src/theory/ff/util.h

```diff
@@ -0,0 +1,85 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Alex Ozdemir
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2023 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * utilities
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__FF__UTIL_H
+#define CVC5__THEORY__FF__UTIL_H
+
+// external includes
+#ifdef CVC5_USE_COCOA
+#include <CoCoA/ring.H>
+#endif /* CVC5_USE_COCOA */
+
+// std includes
+#include <unordered_map>
+
+// internal includes
+#include "expr/node.h"
+#include "util/finite_field_value.h"
+
+namespace cvc5::internal {
+namespace theory {
+namespace ff {
+
+/** A finite field model */
+using FfModel = std::unordered_map<Node, FiniteFieldValue>;
+
+/** The result of a subsolver. */
+enum class Result
+{
+  SAT,
+  UNSAT,
+  UNKNOWN,
+};
+
+/** A class associated with a specific field (for inheritting). */
+class FieldObj
+{
+ public:
+  FieldObj(const FfSize& size);
+  /** create a sum (with as few as 0 elements) */
+  Node mkAdd(std::vector<Node>&& summands);
```

Comment:
Since this is less common, say why an rvalue reference.

Reply:
Good comment. In general, r-value references have some efficiency benefits, but I just took a closer look at `NodeManager::mkNode` and saw an r-value reference is not helpful here. I changed this to be a constant reference to a vector of Nodes or TNodes, which aligns with `NodeManager::mkNode`.